src/qiskit_iqm/fake_backends/iqm_fake_backend.py

```diff
@@ -0,0 +1,185 @@
+
+        # Add two-qubit gate errors to noise model
+        for gate in self.basis_2_qubit_gates:
+            for connection in list(self.chip_sample.two_qubit_gate_depolarization_rates[gate].keys()):
+                first_qubit = connection[0]
+                second_qubit = connection[1]
+
+                thermal_relaxation_channel = thermal_relaxation_error(
+                    self.chip_sample.t1s[first_qubit],
+                    self.chip_sample.t2s[first_qubit],
+                    self.chip_sample.two_qubit_gate_durations[gate],
+                ).tensor(
+                    thermal_relaxation_error(
+                        self.chip_sample.t1s[second_qubit],
+                        self.chip_sample.t2s[second_qubit],
+                        self.chip_sample.two_qubit_gate_durations[gate],
+                    )
+                )
+                depolarizing_channel = depolarizing_error(
+                    self.chip_sample.two_qubit_gate_depolarization_rates[gate][connection], 2
+                )
+                full_error_channel = thermal_relaxation_channel.compose(depolarizing_channel)
+                noise_model.add_quantum_error(full_error_channel, gate, [first_qubit, second_qubit])
+
+        return noise_model
+
+    def run(self, run_input: Union[QuantumCircuit, list[QuantumCircuit]], **options) -> JobV1:
+        """
+        Run `run_input` on the fake backend using a simulator.
+
+        This method runs circuit jobs (an individual or a list of QuantumCircuit
+        ) and returns a :class:`~qiskit.providers.JobV1` object.
+
+        It will run the simulation with a noise model of the fake backend (e.g. Adonis).
+
+        Args:
+            run_input (QuantumCircuit or list of QuantumCircuit): An
+                individual or a list of
+                :class:`~qiskit.circuits.QuantumCircuit` objects to run on the backend.
+            options: Any kwarg options to pass to the backend.
+        Returns:
+            JobV1: The job object for the run
+        Raises:
+            ValueError: If empty list of circuits is provided.
+        """
+        circuits = [run_input] if isinstance(run_input, QuantumCircuit) else run_input
+
+        if len(circuits) == 0:
+            raise ValueError("Empty list of circuits submitted for execution.")
+
+        shots = options.get("shots", self.options.shots)
+
+        # COUPLING MAP
+        coupling_map = CouplingMap(self.qubit_connectivity)
+        # otherwise it would throw an error in the transpiler that Flipping gate direction is not supported for CPhase
+        coupling_map.make_symmetric()
+
+        # Create noisy simulator backend and run circuits
+        sim_noise = AerSimulator(noise_model=self.noise_model, coupling_map=coupling_map, basis_gates=self.basis_gates)
+        job = sim_noise.run(circuits, shots=shots)
+
+        return job
+
+
+class IQMFakeAdonis(IQMFakeBackend):
```

Comment:
Is this subclass necessary? What if the user would just initialize IQMFakeBackend with `adonis_chip_sample`?

Reply:
For convenience reasons and to match Qiskit style I would say yes